packages/dev/core/src/Meshes/meshUtilities.ts

```diff
@@ -0,0 +1,62 @@
+import { AnimationGroup } from "../Animations/animationGroup";
+import { BoundingBox } from "../Culling/boundingBox";
+import { Vector3 } from "../Maths/math.vector";
+import type { AbstractMesh } from "./abstractMesh";
+
+/**
+ * Computes the maximum bounding boxes of the given meshes taking animation, skeleton, morph targets into account.
+ * @param meshes The meshes to compute
+ * @param animationGroup An optional animation group to animate
+ * @param animationStep An optional animation step value indicating the amount to iterate while computing bounding boxes
+ * @returns An array of bounding boxes corresponding to the input meshes, animation group, and animation step values
+ */
+export function computeMaxBoundingBoxes(meshes: Array<AbstractMesh>, animationGroup?: AnimationGroup, animationStep = 1 / 6): Array<BoundingBox> {

```

Comment:
The `meshes` array is not intended to be mutated, yes? Should it a be a `DeepImmutable`?

Reply:
That's a good idea.